wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/InternalRuntime.java

```diff
@@ -197,22 +197,11 @@ public final T getContext() {
             String name = entry.getKey();
             Server incomingServer = entry.getValue();
             try {
-                URL url = new URL(incomingServer.getUrl());
-                ServerImpl server = new ServerImpl(urlRewriter.rewriteURL(identity, name, url).toString(),
+                ServerImpl server = new ServerImpl(urlRewriter.rewriteURL(identity, name, incomingServer.getUrl()),
                                                    incomingServer.getStatus());
                 outgoing.put(name, server);
-            } catch (MalformedURLException e) {
-                warnings.add(new Warning() {
-                    @Override
-                    public int getCode() {
-                        return 101;
-                    }
-
-                    @Override
-                    public String getMessage() {
-                        return "Malformed URL for " + name + " : " + e.getMessage();
-                    }
-                });
+            } catch (InfrastructureException e) {
+                warnings.add(new WarningImpl(101, "Malformed URL for " + name + " : " + e.getMessage()));
```

Comment:
What is _101_?
Have you considered implementing warning for this certain case e.g.
```java
class MalformedURLWarning implements Warning { 
  static int CODE = 101;
  //...
  int getCode() { return CODE; }
  //...
}
```